app/src/main/java/at/team30/setroute/ui/routes/RouteListViewModel.kt

```diff
@@ -21,13 +23,12 @@ class RouteListViewModel @Inject constructor(
 
     init {
         val result = routesRepository.getRoutes()
-        result
         routesList.value = result
     }
 
     fun getRoutes(): LiveData<List<Route>> {
-        routesList.postValue(getSortedRoutes())
-        return routesList
+        routesList.postValue(getSortedRoutes(getFilteredRoutes())) // combination of sorting and filtering sometimes fucks up
+        return routesList                                          // but filtering and sorting on their own work as intended
```

Comment:
The problem (probably) is that this call only defines the initial value of the LiveData Object.

Everytime you either call `applySorting...` or `applyFiltering...` either sorting of filtering will be ignored, since only sorted or filtered routes will be posted.

Should be resolved if you replace the `routeList.postValue(..)` in `applySorting...` and `applyFiltering...` with this exact call you have made here

```
routesList.postValue(getSortedRoutes(getFilteredRoutes()))
```

Reply:
Not really well designed by me. Maybe we can improve this later on, but this quick fix should be enough for now.